### ISSUE TITLE
Maximum call stack, presumably due to my 3100 line long markdown file

### ISSUE THREAD

**Issue by boehs** [pngwn/MDsveX#358]
Don't ask why.

![image](https://user-images.githubusercontent.com/51836263/138012449-23009483-052f-4f9b-ad2f-9082f1d5d586.png)

It's a glorious 138kb file

## Tests I did:

1. Moving the mostrosity of a markdown file to the routes folder
  * this did not work
2. Converting it to HTML before hand and putting it in my routes
  * this did work.

This makes it likely a MDsveX issue

```
9:07:39 PM [vite] Internal server error: Maximum call stack size exceeded
  File: C:/Users/<redacted>/Personal/Notes/blog/<You really don't want to know>.md
      at handle_var_declaration (file:///C:/Users/<redacted>/Personal/Code/Contrib/site/node_modules/.pnpm/svelte@3.44.0/node_modules/svelte/compiler.mjs:6470:10)       
      at VariableDeclaration (file:///C:/Users/<redacted>/Personal/Code/Contrib/site/node_modules/.pnpm/svelte@3.44.0/node_modules/svelte/compiler.mjs:6755:10)
      at handle (file:///C:/Users/<redacted>/Personal/Code/Contrib/site/node_modules/.pnpm/svelte@3.44.0/node_modules/svelte/compiler.mjs:6175:17)
      at file:///C:/Users/<redacted>/Personal/Code/Contrib/site/node_modules/.pnpm/svelte@3.44.0/node_modules/svelte/compiler.mjs:6408:18
      at Array.map (<anonymous>)
      at handle_body (file:///C:/Users/<redacted>/Personal/Code/Contrib/site/node_modules/.pnpm/svelte@3.44.0/node_modules/svelte/compiler.mjs:6407:21)
      at file:///C:/Users/<redacted>/Personal/Code/Contrib/site/node_modules/.pnpm/svelte@3.44.0/node_modules/svelte/compiler.mjs:6489:7
      at scoped_fn (file:///C:/Users/<redacted>/Personal/Code/Contrib/site/node_modules/.pnpm/svelte@3.44.0/node_modules/svelte/compiler.mjs:6376:10)
      at handle (file:///C:/Users/<redacted>/Personal/Code/Contrib/site/node_modules/.pnpm/svelte@3.44.0/node_modules/svelte/compiler.mjs:6175:17)
      at file:///C:/Users/<redacted>/Personal/Code/Contrib/site/node_modules/.pnpm/svelte@3.44.0/node_modules/svelte/compiler.mjs:6748:7
Terminate batch job (Y/N)?
^C
```

**Comment by pngwn** [comment]
I think I know what this is but I don't know how much I can do about it. The files needs to be loaded into memory and the AST needs to exist in memory as well because this mdsvex is primarily a preprocessor. It is probably the size of the AST that is causing an issue.

I'll take a closer look and try to figure out exactly what the problem is, increasing the memory of the node process will probably fix this as a quick workaround.

**Comment by kdheepak** [comment]
I ran into the same issue. If you patch the svelte compiler, it might resolve it for you. See https://github.com/sveltejs/svelte/issues/4694 for more information.

**Comment by benmccann** [comment]
The issue is hopefully fixed now with the release of [Svelte 3.46.4](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md#3464). Could you try again and check if it works now?

**Comment by kdheepak** [comment]
Yes, for me it has!
